concept/type/owns.feature

```diff
@@ -0,0 +1,3952 @@
+      | relation  | description    | registration | decimal       |
+      | relation  | description    | registration | string        |
+      | relation  | description    | registration | boolean       |
+      | relation  | description    | registration | datetime      |
+      | relation  | description    | registration | datetimetz    |
+      | relation  | description    | registration | duration      |
+      | relation  | description    | registration | custom-struct |
+
+  Scenario Outline: <root-type> types cannot redeclare inherited owns as owns
+    When put attribute type: email
+    When attribute(email) set value-type: <value-type>
+    When put attribute type: name
+    When attribute(name) set value-type: <value-type>
+    When <root-type>(<supertype-name>) set owns: name
+    Then <root-type>(<subtype-name>) set owns: name
+    Then transaction commits; fails
+    Examples:
+      | root-type | supertype-name | subtype-name | value-type |
+      | entity    | person         | customer     | string     |
+      | entity    | person         | customer     | long       |
+      | relation  | description    | registration | string     |
+      | relation  | description    | registration | datetime   |
+
+  Scenario Outline: <root-type> types cannot redeclare inherited owns in multiple layers of inheritance
+    When put attribute type: name
+    When attribute(name) set value-type: <value-type>
+    When attribute(name) set annotation: @abstract
+    When put attribute type: customer-name
+    When attribute(customer-name) set value-type: <value-type>
+    When attribute(customer-name) set supertype: name
+    When <root-type>(<supertype-name>) set annotation: @abstract
+    When <root-type>(<supertype-name>) set owns: name
+    When <root-type>(<subtype-name>) set owns: customer-name
+    When <root-type>(<subtype-name-2>) set supertype: <subtype-name>
+    Then <root-type>(<subtype-name-2>) set owns: name
+    Then transaction commits; fails
+    Examples:
+      | root-type | supertype-name | subtype-name | subtype-name-2 | value-type    |
+      | entity    | person         | customer     | subscriber     | datetimetz    |
+      | entity    | person         | customer     | subscriber     | custom-struct |
+      | relation  | description    | registration | profile        | decimal       |
+      | relation  | description    | registration | profile        | string        |
+
+  Scenario Outline: <root-type> types cannot redeclare overridden owns as owns
+    When put attribute type: name
+    When attribute(name) set value-type: <value-type>
+    When attribute(name) set annotation: @abstract
+    When put attribute type: customer-name
+    When attribute(customer-name) set value-type: <value-type>
+    When attribute(customer-name) set supertype: name
+    When <root-type>(<supertype-name>) set annotation: @abstract
+    When <root-type>(<supertype-name>) set owns: name
+    # TODO: No set override here?
```

Comment:
I've additionally left some override `TODO`s as we don't have a clear decision on this feature for `3.0`. I am ready to leave it as is for now and return to it when fixing the logic, I am also ready to refactor these cases now.